simulariumio/orientations/rotation_utility.py

```diff
@@ -0,0 +1,209 @@
+        """
+        if vector[0] == 0 and vector[1] == 0:
+            if vector[2] == 0:
+                raise ValueError("zero vector")
+            return np.array([0, 1, 0])
+        unit_vector = RotationUtility.normalize(np.array([-vector[1], vector[0], 0]))
+        return RotationUtility._rotate(unit_vector, vector, 2 * np.pi * random.random())
+
+    @staticmethod
+    def get_non_periodic_boundary_position(
+        relative_position: np.ndarray, box_size: np.ndarray
+    ) -> np.ndarray:
+        """
+        If the magnitude of a difference in positions is greater than box size,
+        move the it across the box.
+
+        Parameters
+        ----------
+        relative_position: np.ndarray (shape=3)
+            A numpy array of a difference in positions within the box
+        box_size: np.ndarray (shape=3)
+            A numpy array of the box size in X, Y, and Z dimensions
+        """
+        result = np.copy(relative_position)
+        for dim in range(3):
+            if relative_position is None:
+                raise Exception("relative_position is None")
+            if abs(relative_position[dim]) > box_size[dim] / 2.0:
+                result[dim] -= (
+                    relative_position[dim] / abs(relative_position[dim]) * box_size[dim]
+                )
+        return result
+
+    @staticmethod
+    def normalize(vector: np.ndarray) -> np.ndarray:
+        """
+        Normalize a vector.
+
+        Parameters
+        ----------
+        vector: np.ndarray
+            A numpy array to normalize
+        """
+        return vector / np.linalg.norm(vector)
+
+    @staticmethod
+    def _vectors_are_colinear(vector1: np.ndarray, vector2: np.ndarray) -> bool:
+        """
+        Check if two vectors are colinear to each other
+        (either parallel or anti-parallel)
+        """
+        return (
+            abs(
+                np.dot(
+                    RotationUtility.normalize(vector1),
+                    RotationUtility.normalize(vector2),
+                )
+            )
+            >= 1 - float_info.epsilon
+        )
+
+    @staticmethod
+    def get_rotation_matrix_from_bases(
+        vector1: np.ndarray, vector2: np.ndarray
+    ) -> np.ndarray:
+        """
+        Create a rotation matrix given two basis vectors.
+
+        Parameters
+        ----------
+        vector1: np.ndarray
+            A numpy array for the first basis vector (already normalized)
+        vector2: np.ndarray
+            A numpy array for the second basis vector (already normalized)
+        """
+        # cross to get 3rd basis
+        vector3 = np.cross(vector2, vector1)
+        # create matrix with basis
+        return np.array(
+            [
+                [vector1[0], vector2[0], vector3[0]],
+                [vector1[1], vector2[1], vector3[1]],
+                [vector1[2], vector2[2], vector3[2]],
+            ]
+        )
+
+    @staticmethod
+    def get_random_rotation_matrix():
```

Comment:
use scipy rotation?

Reply:
or generate from random euler angles using your get_rotation_matrix_from_euler_angles... anything to simplify this to a one-liner that's super readable.